### PR TITLE
Implement aggregation helpers

### PR DESCRIPTION
**Pull request by terencechain**
This PR implements aggregation helper as specified: https://github.com/ethereum/eth2.0-specs/blob/v0.9.0/specs/validator/0_beacon-chain-validator.md

* slot_signature
* is_aggregator
* get_aggregate_signature

### CODE REVIEW

File: beacon-chain/core/helpers/attestation.go [hunk 1]
```diff
@@ -105,3 +111,57 @@ func AggregateAttestation(a1 *ethpb.Attestation, a2 *ethpb.Attestation) (*ethpb.
...
+}
+
+// IsAggregator returns true if the signature is from the input validator.
+//
+// Spec pseudocode definition:
+//   def is_aggregator(state: BeaconState, slot: Slot, index: CommitteeIndex, slot_signature: BLSSignature) -> bool:
+//    committee = get_beacon_committee(state, slot, index)
+//    modulo = max(1, len(committee) // TARGET_AGGREGATORS_PER_COMMITTEE)
+//    return bytes_to_int(hash(slot_signature)[0:8]) % modulo == 0
+func IsAggregator(state *pb.BeaconState, slot uint64, index uint64, sig *bls.Signature) (bool, error) {
```
Comment (0xKiwi) [inline beacon-chain/core/helpers/attestation.go]:
```suggestion
func IsAggregator(state *pb.BeaconState, slot uint64, index uint64, slotSig *bls.Signature) (bool, error) {
```

File: beacon-chain/core/helpers/attestation_test.go [hunk 2]
```diff
@@ -207,3 +211,59 @@ func TestAggregateAttestations(t *testing.T) {
...
+	agg, err := helpers.IsAggregator(beaconState, 0, 0, sig)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if !agg {
+		t.Error("Wanted aggregator true, got false")
+	}
+}
+
+func TestAggregateSignature_True(t *testing.T) {
```
Comment (0xKiwi) [inline beacon-chain/core/helpers/attestation_test.go]: Can you add a test case for false?

File: beacon-chain/core/helpers/attestation_test.go [hunk 3]
```diff
@@ -207,3 +211,59 @@ func TestAggregateAttestations(t *testing.T) {
...
+	}
+
+	domain := helpers.Domain(state.Fork, helpers.CurrentEpoch(state), params.BeaconConfig().DomainBeaconAttester)
+	msg, _ := ssz.HashTreeRoot(slot)
+	if !sig.Verify(msg[:], pub, domain) {
+		t.Error("Could not verify slot signature")
+	}
+}
+
+func TestIsAggregator_True(t *testing.T) {
```
Comment (0xKiwi) [inline beacon-chain/core/helpers/attestation_test.go]: Can you add a test case for false?

File: shared/params/config.go [hunk 4]
```diff
@@ -26,6 +26,7 @@ type BeaconChainConfig struct {
 	ShuffleRoundCount              uint64 `yaml:"SHUFFLE_ROUND_COUNT"`                // ShuffleRoundCount is used for retrieving the permuted index.
 	MinGenesisActiveValidatorCount uint64 `yaml:"MIN_GENESIS_ACTIVE_VALIDATOR_COUNT"` // MinGenesisActiveValidatorCount defines how many validator deposits needed to kick off beacon chain.
 	MinGenesisTime                 uint64 `yaml:"MIN_GENESIS_TIME"`                   // MinGenesisTime is the time that needed to pass before kicking off beacon chain. Currently set to Jan/3/2020.
+	TargetAggregatorsPerCommittee  uint64 // TargetAggregatorsPerCommittee defines the number of aggregators inside one committee.
```
Comment (0xKiwi) [inline shared/params/config.go]: Does this need a yaml definition?

Comment (terencechain) [reply shared/params/config.go]: nope

File: beacon-chain/core/helpers/attestation.go (hunk 1 above)
Comment (terencechain) [reply beacon-chain/core/helpers/attestation.go]: Sure!

File: beacon-chain/core/helpers/attestation_test.go [hunk 5]
```diff
@@ -207,3 +211,99 @@ func TestAggregateAttestations(t *testing.T) {
...
+		t.Error("Wanted aggregator false, got true")
+	}
+}
+
+func TestAggregateSignature_True(t *testing.T) {
+	pubkeys := make([]*bls.PublicKey, 0, 100)
+	atts := make([]*ethpb.Attestation, 0, 100)
+	msg := []byte("hello")
+	for i := 0; i < 100; i++ {
+		priv, _ := bls.RandKey(rand.Reader)
```
Comment (0xKiwi) [inline beacon-chain/core/helpers/attestation_test.go]: That blank identifier is an error right? Can you change it to an err and handle a failure?

File: beacon-chain/core/helpers/attestation_test.go [hunk 6]
```diff
@@ -207,3 +211,99 @@ func TestAggregateAttestations(t *testing.T) {
...
+		t.Error("Signature did not verify")
+	}
+}
+
+func TestAggregateSignature_False(t *testing.T) {
+	pubkeys := make([]*bls.PublicKey, 0, 100)
+	atts := make([]*ethpb.Attestation, 0, 100)
+	msg := []byte("hello")
+	for i := 0; i < 100; i++ {
+		priv, _ := bls.RandKey(rand.Reader)
```
Comment (0xKiwi) [inline beacon-chain/core/helpers/attestation_test.go]: Same here

File: beacon-chain/core/helpers/attestation_test.go (hunk 5 above)
Comment (terencechain) [reply beacon-chain/core/helpers/attestation_test.go]: Sure, but we do do that everywhere else ¯\_(ツ)_/¯
It's less of an issue for these tests

**Pull request merged**